### PR TITLE
gh-106948: Doc config ignores more standard C functions

### PR DESCRIPTION
**Pull request by vstinner**
Complete nitpick_ignore list in Doc/conf.py to ignore more standard C functions, variables, macros, and also Win32 API functions and macros.

No longer ignore "__int" C type: it should not be used in the doc.

* Issue: gh-106948

----
:books: Documentation preview :books:: https://cpython-previews--107301.org.readthedocs.build/

### CODE REVIEW

**Comment by vstinner** [comment]
@erlend-aasland: In commit b447e19e720e6781025432a40eb72b1cc93ac944 (PR #107062), you decided to not add ``siginterrupt()`` to nitpick_ignore but use ``!siginterrupt`` syntax in the doc.

@serhiy-storchaka wrote that we should not ignore functions which are referenced only once (or twice): https://github.com/python/cpython/pull/107062#issuecomment-1646780385

This PR goes against that. I wrote it before reading PR #107062. I updated nitpick_ignore by going through the whole list of ``sphinx-warnings.txt`` (after I removed all ``Doc/c-api/`` from ``Doc/tools/.nitignore``).

Do you want me to count how many times a function is called to decide to ignore it or not?

I skipped many functions which don't belong to the C library (libc), but readline or other library. I also skipped ``getopt()`` for example. I would prefer to use ``!function()`` syntax in the doc for these instead.

File: Doc/conf.py [hunk 1]
```diff
@@ -78,43 +78,119 @@
...
     ('c:type', 'ptrdiff_t'),
     ('c:type', 'siginfo_t'),
     ('c:type', 'size_t'),
     ('c:type', 'ssize_t'),
     ('c:type', 'time_t'),
+    ('c:type', 'uint64_t'),
     ('c:type', 'uintmax_t'),
+    ('c:type', 'uintptr_t'),
     ('c:type', 'va_list'),
+    ('c:type', 'wchar'),
```
Comment (serhiy-storchaka) [inline Doc/conf.py:164]: It is a typo, not a c:type. :wink:

Comment (vstinner) [reply Doc/conf.py:164]: Oh! I wasn't sure about this one, ok, I will remove it. First, I'm trying to land PR #107302 to reduce the number of warnings.

Comment (erlend-aasland) [reply Doc/conf.py:164]:
```suggestion
    ('c:type', 'wchar_t'),
```

**Comment by erlend-aasland** [comment]
I share the same view as Serhiy. IMO, it makes sense to keep only the most common functions (C stdlib stuff) in this list, and use the `!func` markup for rarer functions and syscalls. A larger list will pollute `conf.py`, and as the docs change, it will inevitably contain more and more redundant entries.

I won't block this PR, but I won't approve it either ;)

File: Doc/conf.py (hunk 1 above)
Comment (CAM-Gerlach) [reply Doc/conf.py:164]: Note that `wchar_t` is already listed right below, so the change here should be just removing it:

```suggestion
```

**Review by CAM-Gerlach — COMMENTED** [review]
This LGTM, but I'm not a C expert like the other folks here, so while I gave the list a skim and didn't spot any super-obvious false negatives, I could be easily be missing some.

I do share Erlend's and Serhiy's concerns about going overboard with an overly lengthy list that may contain other undetected mistakes that may in turn silence potentially valid errors. 

On the other hand, while I'm a stickler for correctness and explicitness, that's balanced with the desire to reduce the burden on doc authors having to manually silence warnings to satisfy the CI, which annoys contributors, consumes non-trivial nitpickiness budget, reduces warning SNR and builds a habit of automatically silencing them rather than carefully considering how to handle each of them individually.

For me, its a trade between the chance of a false negative vs. the accumulated author time and annoyance dealing with the false positives the additional names would have prevented (unless they are somehow going to change frequently, I don't see _that_ much real-world ongoing maintenance cost keeping them in `conf.py`).

Personally, I'd rather see a few false negatives and have the time saved silencing false positives spent fixing true positives properly (instead of just habitually silencing them)—as you've shown admirable care about so far in #107302 . @vstinner , if you have (or can) checked them over again fairly carefully, then it seems to me that the benefit of the latter would likely outweigh any impact of the former, so I'd be +0.5 on this.

**Comment by vstinner** [comment]
Ok, I will try this approach. I started with PR #107329 which uses a few ``!func()`` syntax on standard C functions like memcpy() or pthread_atfork().

**Pull request closed without merging**